functions/src/winloss.ts

```diff
@@ -46,19 +46,19 @@ export async function endGameProcess(
   transaction: firestore.Transaction
 ) {
   // apply the stats
-  let { players } = lobbyData;
-  const { uids, winner } = lobbyData;
+  const { players } = lobbyData;
+  const { winner } = lobbyData;
 
   // update each players doc
   await Promise.all(
-    uids.map(async (uid, index) => {
+    Object.keys(lobbyData.players).map(async (uid) => {
```

Comment:
consider Object.entries instead of keys so you don't have to do lobbyData.players[uid] a bunch